cmd/juju/caas/add.go

```diff
@@ -228,6 +282,127 @@ func (c *AddCAASCommand) Run(ctxt *cmd.Context) error {
 	return nil
 }
 
+func (c *AddCAASCommand) addCloudToControllerWithRegion(apiClient AddCloudAPI, newCloud jujucloud.Cloud) (err error) {
+	if newCloud.HostCloudRegion != "" {
+		if err = c.validateCloudRegion(newCloud.HostCloudRegion); err != nil {
+			return errors.Trace(err)
+		}
+	}
+	if err := addCloudToController(apiClient, newCloud); err != nil {
+		return errors.Trace(err)
+	}
+	return nil
+}
+
+func newK8sBrokerGetter(rootAPIGetter func() (api.Connection, error)) BrokerGetter {
+	return func(cloud jujucloud.Cloud, credential jujucloud.Credential) (k8sBrokerRegionLister, error) {
+		conn, err := rootAPIGetter()
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		modelAPI := modelconfig.NewClient(conn)
+		defer modelAPI.Close()
+
+		// this current model is not really we want, but it's just for generating the config.
```

Comment:
We can reword this comment.
// Use the controller model config for constructing the Juju k8s client.